plugins/network-elements/nuage-vsp/src/com/cloud/network/guru/NuageVspGuestNetworkGuru.java

```diff
@@ -129,66 +132,93 @@ public Network design(NetworkOffering offering, DeploymentPlan plan, Network use
 
     @Override
     public Network implement(Network network, NetworkOffering offering, DeployDestination dest, ReservationContext context) throws InsufficientVirtualNetworkCapacityException {
+        long networkId = network.getId();
+        network = _networkDao.acquireInLockTable(network.getId(), 1200);
+        if (network == null) {
+            throw new ConcurrentOperationException("Unable to acquire lock on network " + networkId);
+        }
+
+        NetworkVO implemented = null;
+        try {
+            assert (network.getState() == State.Implementing) : "Why are we implementing " + network;
```

Comment:
Personally, I love assertions.  However, as a project, we don't consistently run with assertions enabled.  Additionally, the incomplete test coverage does not guarantee that the assertion will properly exercised.  Therefore, I suggest converting into an ``if`` block and throw an ``IllegalStateException`` to ensure that the conditions is always checked.